### PR TITLE
Close JournalServiceClient asynchronously to account for async calls

### PR DESCRIPTION
**Pull request by jenoudet**
Currently, the `JournalServiceClient` used in `SnapshotReplicationManager` is closed using try-resource blocks. This causes the async calls within said blocks to fail sometimes as the client is closed when the call is launched.
This PR ensures that the clients are closed asynchronously after the call is finished using `whenComplete` blocks. 
A test accompanies this to ensure proper snapshot propagation.

### CODE REVIEW

File: core/server/common/src/main/java/alluxio/master/journal/raft/SnapshotReplicationManager.java [hunk 1]
```diff
@@ -519,7 +524,7 @@ private boolean requestData() {
   }
 
   @VisibleForTesting
-  synchronized RaftJournalServiceClient getJournalServiceClient()
+  synchronized RaftJournalServiceClient createJournalServiceClient()
```
Comment (jenoudet) [inline core/server/common/src/main/java/alluxio/master/journal/raft/SnapshotReplicationManager.java:527]: This method was renamed after Rongrong's presentation on Java anti-patterns, notably functions called `get` that are not getters.

File: minicluster/src/main/java/alluxio/multi/process/PortCoordination.java [hunk 2]
```diff
@@ -45,7 +46,6 @@ public class PortCoordination {
   // for EmbeddedJournalIntegrationTestTransferLeadership
   public static final List<ReservedPort> EMBEDDED_JOURNAL_TRANSFER_LEADER = allocate(5, 0);
   public static final List<ReservedPort> EMBEDDED_JOURNAL_REPEAT_TRANSFER_LEADER = allocate(5, 0);
-  public static final List<ReservedPort> EMBEDDED_JOURNAL_RESET_PRIORITIES = allocate(5, 0);
```
Comment (jenoudet) [inline minicluster/src/main/java/alluxio/multi/process/PortCoordination.java:48]: Removed because it was unused.

**Comment by ggezer** [comment]
alluxio-bot, merge this please

**Comment by jenoudet** [comment]
@Xenorith cherry pick this please

**Comment by Xenorith** [comment]
alluxio-bot, cherry-pick this to branch-2.8 please

**Pull request merged**